### ISSUE TITLE
API calls & session handling (e.g. user authentication)

### ISSUE THREAD

**Issue by afterburn** [remarkablemark/tutorials#9]
Lately there has been alot of buzz regarding Vue.js. Some claim it's even a React killer so I thought it'd be nice to have both sides of the story with regard to SSR and creating universal apps and I think you'd be excellent at conveying this. You've already done a series on universal react applications, so I was hoping maybe you could now do a similar series on Vue.

[Vue](https://vuejs.org/)
[Vuex](https://vuex.vuejs.org/en/)
[Vue SSR](https://vuejs.org/v2/guide/ssr.html)

**Comment by remarkablemark** [comment]
Thanks for the tutorial request @afterburn!

Unfortunately, I may not get to making video tutorials on Vue because of my unfamiliarity with the framework and my current backlog of work. Nevertheless, I can prototype something quickly and write what I learned (if that helps).

Were you able to build a universal app with Vue? How did that go?

Also, I found a resource that you may interesting:
- [Egghead tutorials on Vue](https://egghead.io/courses/develop-web-apps-with-vue-js)

**Comment by afterburn** [comment]
Hi @remarkablemark, thanks for your reply. I understand completely, unfortunately no I was not able to build an universal app with Vue but it got my attention because of it's server-side rendering capabilities.

I'm trying to basically create an app that lets users log in and have them be able to post articles, which in turn should be able to be indexed by crawlers for SEO purposes. Of course this is simple to accomplish in say PHP, but I would really like to have some kind of SPA functionality for my users (Angular, React, Vue). Naturally I found your universal react videos very informative. 

I however cannot wrap my head around how to now implement state management. I was hoping maybe you could continue your Universal React App series and implement something like [MobX](https://github.com/mobxjs/mobx), [Redux ](http://redux.js.org) or [Flux](https://facebook.github.io/flux/) because I feel this would really give me all the information I need to create such an app.

Kind regards.

**Comment by afterburn** [comment]
To really cover everything there is to creating a Universal React application (imho) perhaps it is wise to have a video in which you show us how to add a couple of new routes to the app you've created in the series so far.

**Comment by afterburn** [comment]
Nevermind, I just found out you've already done videos about them haha. Amazing work Mark.
Perhaps as a continuation of this series you could some tutorials on how to now implement an API on the server-side. I'm curious to see how you would approach Authentication and managing sessions in this Universal React application.

**Comment by remarkablemark** [comment]
Great suggestions @afterburn.

The last [video](https://b.remarkabl.org/2hybtYR) in my universal React series goes over how to set up Redux (but the reducer and actions aren't fleshed out yet).

I do plan on going over sessions, but I'm blocked by #1. Progress is slow because of work and other projects, but I'll try my best to finish them in my spare time.

Thanks as always for understanding and being patient!
